include/odp/api/spec/packet.h

```diff
@@ -1799,15 +1799,23 @@ uint32_t odp_packet_flow_hash(odp_packet_t pkt);
  * Store the packet flow hash for the packet and sets the flow hash flag. This
  * enables (but does not require!) application to reflect packet header
  * changes in the hash.
+ * When scheduler is configured as flow aware, schedule queue synchronization
+ * will be based on flow within each queue.
+ * When scheduler is configured as flow unaware, packet flow hash is ignored by
+ * the implementation.
+ * The value of flow hash should not be greater than the max flow count
+ * supported by the implementation.
```

Comment:
It would be better to rewrite this description entirely rather than to insert this addendum as the original text refers to behavior in flow unaware mode and then this describes both flow aware and unaware modes again.

This is a good place to provide an overall discussion of what a flow hash represents in the two modes and how ODP will interpret these values.

Reply:
The wording here also makes it unclear that (at least in the Cavium implementation) the 32-bit hash value is divided into some number of bits designating a queue with the remaining bits representing the flow within that queue. We certainly need to make this implementation use case explicit. Whether there are other possible implementation-specific interpretations of the hash field is something that still needs to be discussed, preferably with other HW implementations to see if there are other variants the spec really should be addressing.

Reply:
Bill, only flow hash part is saved into packet. 32 bit hash value in HW is combined from (bottom) bits of this hash and the queue ID. Here we need to specify just which bits/values of this uint32_t are considered as flow ID. Queue ID is stored as odp_queue_t as before.